libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/transform_exclusive_scan.hpp

```diff
@@ -0,0 +1,563 @@
+    ///           \a parallel_task_policy and
+    ///           returns \a O otherwise.
+    ///           The \a transform_exclusive_scan algorithm returns the output
+    ///           iterator to the element in the destination range, one past
+    ///           the last element copied.
+    ///
+    /// \note   GENERALIZED_NONCOMMUTATIVE_SUM(op, a1, ..., aN) is defined as:
+    ///         * a1 when N is 1
+    ///         * op(GENERALIZED_NONCOMMUTATIVE_SUM(op, a1, ..., aK),
+    ///           GENERALIZED_NONCOMMUTATIVE_SUM(op, aM, ..., aN)
+    ///           where 1 < K+1 = M <= N.
+    ///
+    /// Neither \a conv nor \a op shall invalidate iterators or subranges, or
+    /// modify elements in the ranges [first,last) or [result,result +
+    /// (last - first)).
+    ///
+    /// The behavior of transform_exclusive_scan may be non-deterministic for
+    /// a non-associative predicate.
+    ///
+    template <typename ExPolicy, typename Rng,  typename O, typename T,
+        typename BinOp, typename UnOp>
+    typename parallel::util::detail::algorithm_result<ExPolicy, O>::type
+    transform_exclusive_scan(ExPolicy&& policy, Rng&& rng, O dest, T init,
+        BinOp&& binary_op, UnOp&& unary_op);
+
+    // clang-format on
+}}    // namespace hpx::ranges
+#else
+
+#include <hpx/config.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
+#include <hpx/executors/execution_policy.hpp>
+#include <hpx/functional/tag_fallback_dispatch.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/parallel/algorithms/transform_exclusive_scan.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/projection_identity.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx { namespace ranges {
+    HPX_INLINE_CONSTEXPR_VARIABLE struct transform_exclusive_scan_t final
+      : hpx::functional::tag_fallback<transform_exclusive_scan_t>
+    {
+    private:
+        // clang-format off
+        template <typename InIter, typename Sent, typename OutIter,
+             typename T, typename BinOp, typename UnOp,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter>::value &&
+                hpx::traits::is_sentinel_for<Sent, InIter>::value &&
+                hpx::traits::is_iterator<OutIter>::value &&
+                hpx::is_invocable_v<UnOp,
+                    typename std::iterator_traits<InIter>::value_type> &&
+                hpx::is_invocable_v<BinOp,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<InIter>::value_type>::type,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<InIter>::value_type>::type
+                >
+            )>
+        // clang-format on
+        friend OutIter tag_fallback_dispatch(
```

Comment:
Same as for the other scan algorithms: should we return `in_out_result<InIter, OuIter>` instead?